module-system/sov-state/src/storage.rs

```diff
@@ -160,6 +160,17 @@ pub trait Storage: Clone {
         + BorshSerialize
         + BorshDeserialize;
 
+    /// A cryptographic commitment to the contents of this storage
+    type Root: Serialize
```

Comment:
Nit: just note on naming. For me it looks like this is more "Hash", because root_hash is hash in the end, but maybe hash is too broad as well.

Reply:
Hmm, Hash isn't quite accurate here because the root can be something other than a Hash (I.e. a curve point). Also, it's a bit overloaded - it might be confused with the "Hasher" generic